File: server/php/src/Application/Extensions/Models/LanguageExtensions.php
<?php

namespace OnlinePayments\ExampleApp\Application\Extensions\Models;

use OnlinePayments\ExampleApp\Application\Domain\Enums\Common\Country;
use OnlinePayments\ExampleApp\Application\Domain\Enums\Common\Language;

class LanguageExtensions
{
    private const LANGUAGE_ISO_CODES = [
        Language::English->value => 'en',
        Language::German->value => 'de',
        Language::French->value => 'fr',
    ];

    public static function toLocale(?Language $language, ?Country $country): string
    {
        if ($language === null || $country === null) {
            return '';
        }

        $languageCode = self::LANGUAGE_ISO_CODES[$language->value] ?? '';
        if ($languageCode === '') {
            return '';
        }

        $countryCode = CountryExtension::toIsoAlpha2($country);
        if ($countryCode === '') {
            return '';
        }

        return "{$languageCode}-{$countryCode}";
    }
}

File: server/dotnet/sdk-dotnet-example/Infrastructure/Extensions/Models/LanguageExtensions.cs
using Business.Domain.Common.Enums;

namespace Infrastructure.Extensions.Models;

public static class LanguageExtensions
{
    private static readonly Dictionary<Language, string> LanguageIsoCodes = new()
    {
        { Language.English, "en" },
        { Language.German, "de" },
        { Language.French, "fr" }
    };
    
    public static string ToLocale(this Language? language, Country? country)
    {
        if (!language.HasValue || !country.HasValue)
        {
            return string.Empty;
        }

        if (!LanguageIsoCodes.TryGetValue(language.Value, out var languageCode))
        {
            return string.Empty;
        }

        string? countryCode = country.ToIsoAlpha2();
        
        return string.IsNullOrEmpty(countryCode) ? string.Empty : $"{languageCode}-{countryCode}";
    }
}